### ISSUE TITLE
Add a way to format on save if you are using elixir >= 1.6

### ISSUE THREAD

**Issue by rockneurotiko** [tonini/alchemist.el#337]
Elixir 1.6 is released! And it have the awesome formatter: https://github.com/elixir-lang/elixir/blob/v1.6.0/CHANGELOG.md

I think that it would be a great idea to be able to configure alchemist to format on save.

Cheers!

**Comment by dylnclrk** [comment]
Is there more complexity here than just adding an `after-save-hook` that runs `mix format` in the project root? (e.g., similar to how [`alchemist-hooks-test-on-save`][alchemist-hooks-test-on-save] is implemented). 

Based on what I see in [mix-format.el][mix-format-el], I'm guessing that there's some work to make sure all of the open buffers are updated to include changes from `mix format`.

BTW, @rockneurotiko you should consider checking out https://github.com/anildigital/mix-format.el, I haven't tried it out yet…

[alchemist-hooks-test-on-save]: https://github.com/tonini/alchemist.el/blob/4beb4c9ceee0534a36f7669aa35fcbdfd6bef7b7/alchemist-hooks.el#L49

[mix-format-el]: https://github.com/anildigital/mix-format.el

**Comment by rockneurotiko** [comment]
Yeah, it's an easy feature (Actually I made my own solution 3 months ago, not as good as mix-format but it works fine).

I just thinks that it would be great to have this official formatting added to this de-facto package for elixir, it would probably help newcomers to appreciate the formatter and write clean elixir (it happens on Go and Elm)
